sql/2021/security/feature_adoption_by_technology.sql

```diff
@@ -0,0 +1,43 @@
+#standardSQL
+# Adoption of features based on the technology that is used
+SELECT
+  client,
+  category,
+  app,
+  headername,
+  COUNT(DISTINCT url) AS total_pages_with_technology,
+  COUNT(DISTINCT IF(STARTS_WITH(url, 'https'), url, NULL)) AS total_https_pages,
+  COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername, ' ')), url, NULL)) AS freq,
+  SAFE_DIVIDE(COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername, ' ')), url, NULL)), COUNT(DISTINCT url)) AS pct,
+  SAFE_DIVIDE(COUNT(DISTINCT IF(REGEXP_CONTAINS(respOtherHeaders, CONCAT('(?i)', headername, ' ')) AND STARTS_WITH(url, 'https'), url, NULL)), COUNT(DISTINCT IF(STARTS_WITH(url, 'https'), url, NULL))) AS pct_https
+FROM (
+  SELECT
+    t._TABLE_SUFFIX AS client,
+    category,
+    app,
+    respOtherHeaders,
+    r.urlShort AS url,
+    firstHtml
+  FROM
+    `httparchive.summary_requests.2021_07_01_*` AS r
+  INNER JOIN
+    `httparchive.technologies.2021_07_01_*` AS t
+  ON r._TABLE_SUFFIX = t._TABLE_SUFFIX AND r.urlShort = t.url
```

Comment:
Hmmm... I'm not sure on this join on `urlShort`.

It is convenient to not have to map the `pageid` to the real page url (IMHO the `summary_request` table should have the page url) with another and, in theory, the page should always have a request matching that URL, but still  not 100% sure we should be doing this...

@paulcalvano / @rviscomi what's your thoughts here?

Reply:
Discussed with  @rviscomi and he thinks there might be an issue with redirects sometimes where the URL stored in the `requests`/`summary_requests` tables are not the same as initial origin page. Probably small numbers.

So can you change this to a three way join just to be 100% sure:
`summary_requests` -> (page id) -> `summary_pages` -> (url) -> technologies